Check command buffer callback queue before VkDevice is destroyed

cmannett85-arm opened the pull request:
We only check for signaled events on each queue submit, so the last submission before exit will never be checked.  This change makes sure we check for completed submissions just before the device is destroyed (typically target application exit).

Without this change you will get VVL warnings due to the `VkEvent` not being destroyed before the `VkDevice` - plus anything else that the callbacks were intended to destroy.

baldurk reviewed (changes requested):
I think since this is happening during shutdown we should just destroy these events and close without calling the callbacks, as it seems safer to not call callbacks that may want to do work that's invalid while shutting down - rather than callbacks which _must_ be called.

baldurk commented on renderdoc/driver/vulkan/vk_core.cpp:
```diff
@@ -783,6 +783,23 @@ void WrappedVulkan::CheckPendingCommandBufferCallbacks()
+void WrappedVulkan::ClearPendingCommandBufferCallbacks()
+{
+  // This should be called after a FlushQ() call so that all events have signalled, otherwise the
+  // VkPendingSubmissionCompleteCallbacks entry will leak
```
Minor nit, but since this code should exclusively only be done during shutdown and should never be re-used elsewhere I'd rather just in-line this at the callsite. Rather than having a function which can't and shouldn't be re-used.

cmannett85-arm committed:
Check command buffer callback queue before VkDevice is destroyed

We only check for signaled events on each queue submit, so the last submission before exit will never be checked.  This change makes sure we release for completed submissions just before the device is destroyed (typically target application exit).  In this case the callbacks are not called.
